master/buildbot/schedulers/base.py

```diff
@@ -122,6 +122,14 @@ def getPendingBuildTimes(self):
     @defer.inlineCallbacks
     def startConsumingChanges(self, fileIsImportant=None, change_filter=None,
                               onlyImportant=False):
+        # wait for 10s to start consuming changes
```

Comment:
it's the patch with the [DO NOT MERGE] prefix, it ensures the race condition is reproduced (and that this fix corrects it too). (once done that patch will be removed from the history)